### PR TITLE
updated constants.ts and Heimguard tech.

### PR DESCRIPTION
**Pull request by nagydavid**
Added fingerprint constant under ID:4 to the action_source.

Updated Heimguard Technologies HC-SLM-1 so the action source, user visible, plus pincode converter added as well.

### CODE REVIEW

File: src/devices/heimgard_technologies.ts [hunk 1]
```diff
@@ -23,7 +34,14 @@ const definitions: Definition[] = [
             await endpoint.read('closuresDoorLock', ['lockState', 'soundVolume']);
         },
         exposes: [
-            e.lock(), e.battery(), e.auto_relock_time().withValueMin(0).withValueMax(3600), e.sound_volume()],
+            e.lock(),
+            e.battery(),
+            e.auto_relock_time().withValueMin(0).withValueMax(3600),
+            e.sound_volume(),
+            e.lock_action_user(),
+            e.lock_action_source_name(),
```
Comment (Koenkk) [inline src/devices/heimgard_technologies.ts:42]: Please don't change the formatting

File: package.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
     "@types/tar-stream": "^2.2.2",
     "@typescript-eslint/eslint-plugin": "^5.61.0",
     "@typescript-eslint/parser": "^5.61.0",
-    "eslint": "^8.44.0",
+    "eslint": "^8.45.0",
```
Comment (Koenkk) [inline package.json:54]: Please revert the changes done to package-lock.json and package.json

Comment (nagydavid) [reply package.json]: reverted

File: src/devices/heimgard_technologies.ts (hunk 1 above)
Comment (nagydavid) [reply src/devices/heimgard_technologies.ts]: fixed formating request

**Comment by Koenkk** [comment]
Thanks!

**Comment by rophome** [comment]
Any plans to add this to HC-SLM-2?

**Comment by nagydavid** [comment]
I don't own HC-SLM-2, so I won't be able to test that.
